meta-lmp-bsp/dynamic-layers/stm-st-stm32mp/recipes-support/stm32-mfgtool-files/stm32-mfgtool-files/provision.sh.in

```diff
@@ -0,0 +1,142 @@
+#!/bin/sh
+#
+# Copyright (C) 2022 Foundries.IO
+#
+# SPDX-License-Identifier: MIT
+#
+
+## $1 = publicKeyhash.bin location
+## $2 = key type (dev, prod)
+pubKeyFile=${1}
+keyType=${2}
+
+if [ ! -f ${pubKeyFile} ]; then
+	echo "STM32 publicKeyhash file not found!"
+	echo 1
+fi
+
+# TODO: implement keyType in secure storage
+if [ "${keyType}" != "dev" ] && [ "${keyType}" != "prod" ]; then
+	echo "Invalid STM32 key type!  Must be dev or prod."
+	echo 1
+fi
+
+## VALIDATE STM32 pubKeyFile
+
+pubkeywords=$(hexdump -e '1/4 "%08x" " "' ${publicKeyFile})
+echo "pubkey: ${pubkeywords}"
+
+pubkeyword_count=$(echo "${pubkeywords}" | wc -w)
+echo "count: ${pubkeyword_count}"
+
+if [ "${pubkeyword_count}" -ne "8" ]; then
+	echo "Bad pub key file.  Incorrect length!"
+	exit 1
+fi
+
+
+# FACTORY SETTINGS: .fiodata
+# This file format should be:
+# key=value
+echo "factory_keys=${keyType}" > .fiodata
+
+# export UTC time
+echo -n $(date --utc +"%Y-%m-%d %H:%M:%S") > .run.utc.dat
+
+# start USB download mode and flash primary eMMC partitions
+STM32_Programmer_CLI -c port=USB1 -d flashlayouts-stm32mp1/FlashLayout_emmc_@@BOARD_NAME@@-optee.tsv
```

Comment:
Can we add a check for STM32_Programmer_CLI as part of this script?

It's not setting -e so it will just fail and move on if not found.